src/Microsoft.TestPlatform.CommunicationUtilities/TestRequestSender.cs

```diff
@@ -453,6 +453,28 @@ private void OnExecutionMessageReceived(object sender, MessageReceivedEventArgs
 
                         this.channel.Send(data);
                         break;
+
+                    case MessageType.AttachDebuggerToProcess:
+#pragma warning disable SA1119 // Statement must not use unnecessary parenthesis
+                        if (!(testRunEventsHandler is ITestRunEventsHandler2 handler))
```

Comment:
consider: Invert the condition to avoid negation (and `#pragma`) and put the throw outside the `if` block or in an `else` block instead...

Reply:
Done.